plugins/modules/nd_service.py

```diff
@@ -0,0 +1,146 @@
+author:
+- Sabari Jaganathan (@sajagana)
+options:
+  import_url:
+    description:
+    - The remote location of the Service Package.
+    aliases: [ url ]
+    type: str
+  import_id:
+    description:
+    - The ID of the imported Service Package.
+    aliases: [ id ]
+    type: str
+  state:
+    description:
+    - Use C(present) for importing a Service Package.
+    - Use C(query) for listing all Service Packages.
+    - Use C(absent) for deleting a Service Package.
+    type: str
+    choices: [ present, query, absent ]
+    default: present
+extends_documentation_fragment: cisco.nd.modules
+"""
+
+EXAMPLES = r"""
+- name: Import a service package
+  cisco.nd.nd_service:
+    import_url: "https://nd_service.cisco.com/cisco-terraform-v0.1.16.aci"
+    state: present
+
+- name: Query a service package with import_url
+  cisco.nd.nd_service:
+    import_url: "https://nd_service.cisco.com/cisco-terraform-v0.1.16.aci"
+    state: query
+  register: query_result
+
+- name: Query a service package with import_id
+  cisco.nd.nd_service:
+    import_id: "firmware::v1::ServicePackageImport::e5c480e0-71b2-4ccb-9bb0-b9c174a5731b"
+    state: query
+  register: query_result
+
+- name: Query all service packages
+  cisco.nd.nd_service:
+    state: query
+  register: query_results
+
+- name: Remove a service package with import_id
+  cisco.nd.nd_service:
+    import_id: "firmware::v1::ServicePackageImport::e5c480e0-71b2-4ccb-9bb0-b9c174a5731b"
+    state: absent
+
+- name: Remove a service package with import_url
+  cisco.nd.nd_service:
+    import_url: "https://nd_service.cisco.com/cisco-terraform-v0.1.16.aci"
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        import_url=dict(type="str", aliases=["url"]),
+        import_id=dict(type="str", aliases=["id"]),
+        state=dict(type="str", default="present", choices=["present", "query", "absent"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["import_url"]],
+            ["state", "absent", ["import_id", "import_url"], True],
+        ],
+        mutually_exclusive=[
+            ("import_url", "import_id"),
+        ],
+    )
+
+    nd = NDModule(module)
+
+    import_url = nd.params.get("import_url")
+    import_id = nd.params.get("import_id")
+    state = nd.params.get("state")
+
+    base_path = "/nexus/infra/api/firmware/v1/servicepackageimports"
+    if import_id:
+        # Query a object with meta id
+        service_package = nd.query_obj("{0}/{1}".format(base_path, import_id), ignore_not_found_error=True)
+        if service_package:
+            nd.existing = service_package
+    elif state == "absent" or state == "query":
```

Comment:
why do we not do this for present state?

Reply:
Because some of the service package installations may fail due to Internet connectivity or some other permission issues. So we should allow a user to reinstall the service package from the same remote location.